docs/react-v9/contributing/dev-workflow.md

```diff
@@ -1,4 +1,94 @@
-// TODO #30642 A doc explaining day to day dev work flow.
+## Making changes
 
-Assignee - Mason
-Reviewer - Sarah
+### Creating a branch
+
+Create a branch from your fork for your code changes:
+
+```
+git checkout -b my-branch-name
+```
+
+We strongly recommend using the CLI as your primary interface with git. GUIs are useful for viewing diffs, creating branches and making quick commits. However they often do not correctly merge and sync working branches with master. Mistakes in this step are time consuming to fix.
+
+That said, some team members use [SourceTree](https://www.sourcetreeapp.com/), [GitHub app](https://desktop.github.com/) or the git integrations available in VS Code.
+
+We don't have an official branch naming policy. However some developers use the following to make their branches easier to find: user/[alias]/you-fancy-branch-name, ie user/johndoe/fix-that-bug
+
+### Building
+
+The inner development loop usually involves these steps:
+
+First, `cd` to the root folder of the repo (usually `fluentui`) and run `yarn` to install dependencies and link packages.
+
+```
+cd fluentui
+yarn // this can take some time
+```
+
+Run `yarn start` and select your start up project.
+
+- @fluentui/public-docsite starts the public v8 documentation site.
+- @fluentui/public-docsite-v9 starts the internal v9 documentation storybook site.
+- Pick a specific component package if you intend to work on a v9 component directly.
+
+Most projects have hot module reloading, so file saves will trigger refreshes.
+
+### Iterating
+
+Make the changes you need, and commit along the way.
+
+```
+git status // Shows all changed files
+git add . // Stages the changed files
+git commit -m "Your brief message." // Makes the commits. You may notice a slight lag during committing as our linters work away.
+git push upstream // Pushes your changes to the central Fluent UI repo.
+```
```

Comment:
I think either here or as a separate section we could throw in a few words about running tests. I think the two items I can think of that would be helpful are:
- `yarn workspace @fluentui/package-name test`
- to just test one component, `yarn just jest ComponentName` from within the package, or e.g. `yarn workspace @fluentui/react-combobox just jest Option` from the root

Reply:
Normally I'd agree with you.

However I've run tests locally, then had them faceplant on the CI build, so I've stopped wasting my time running them locally all together.

It's on my mental todo list to add more docs about our high level testing strategy, our required testing bars, and some how tos. 

That's gunna take a lot more investigative legwork on my part. But dream big, start small.

Reply:
I think an additional doc about testing makes sense 👍